File: mqtt-rpc-java/src/test/java/com/ispa/rpc/mqttrpc/CalculatorService.java
/*******************************************************************************
 *******************************************************************************/
package com.ispa.rpc.mqttrpc;

/**
 * @author Brandon Sheffield - 
 * Created on  Jan 5, 2016
 * Description:  Methods below are represented as a Service to a Client to call remotely.
 */
public interface CalculatorService {
	
	public Integer add(Integer a, Integer b);
	public Integer sub(Integer a, Integer b);
	public Integer mult(Integer a, Integer b);

}

File: mqtt-rpc-nodejs/examples/CalculatorService.js
/*******************************************************************************
 *******************************************************************************/
var debug = require('debug')('calculator');

var CalculatorService = function(){
    
    var self = this;


    this.add = function(a,b){
	debug('add',a,b);
	return a + b;
    };

    this.sub = function(a,b){
	debug('sub',a,b);
	return a - b;
    }
};

module.exports = CalculatorService;
